subprojects/core/src/main/java/org/gradle/internal/buildevents/BuildExceptionReporter.java

```diff
@@ -352,11 +353,23 @@ private void fillInFailureResolution(FailureDetails details, ProblemLookup probl
             addBuildScanMessage(context);
         }
 
-        if (!hasNonGradleSpecificCauseInAncestry) {
+        if (shouldDisplayGenericResolutions) {
             context.appendResolution(this::writeGeneralTips);
         }
     }
 
+    private static boolean hasProblemReportsWithSolutions(List<? extends Throwable> throwables, ProblemLookup problemLookup) {
+        for (Throwable throwable : throwables) {
+            Optional<String> first = problemLookup.findAll(throwable).stream().flatMap(p -> p.getSolutions().stream()).findFirst();
+            if (first.isPresent()) {
```

Comment:
❓ Do we really want to only work on the first element in the `throwables` list?

Reply:
That would be incorrect. The method does something less though. It returns `true` if there's at least one solution listed in problems associated with the target exceptions.

Reply:
Why are we recursively calling the method though?
It still means we will never check for the second element in the list.
Or in other words we assume there is never as solution attached to any element in the throwable list except the first one. That can be true for the original invocation, since it's a single element list.
But are we sure it is true for the recursive call?